llvm/test/CodeGen/AMDGPU/barrier-elimination-gfx12.ll

```diff
@@ -0,0 +1,55 @@
+; RUN: llc -global-isel=0 -mtriple=amdgcn -mcpu=gfx1200 < %s | FileCheck %s
+; RUN: llc -global-isel=1 -mtriple=amdgcn -mcpu=gfx1200 < %s | FileCheck %s
+
+; CHECK-LABEL: {{^}}signal_unknown_wgs:
+; CHECK: s_barrier_signal
+define amdgpu_kernel void @signal_unknown_wgs() {
+  tail call void @llvm.amdgcn.s.barrier.signal(i32 -1)
+  ret void
+}
+
+; CHECK-LABEL: {{^}}signal_flat_wgs_attr_32_128:
+; CHECK: s_barrier_signal
+define amdgpu_kernel void @signal_flat_wgs_attr_32_128() #1 {
+  tail call void @llvm.amdgcn.s.barrier.signal(i32 -1)
+  ret void
+}
+
+; CHECK-LABEL: {{^}}signal_flat_wgs_attr_16_32:
+; CHECK: :
+; CHECK-NEXT: s_endpgm
```

Comment:
Nit: Would it be more robust to `CHECK-NOT: barrier` instead?

Reply:
Good point - regenerated check lines with update_llc_test_checks.py to make it more robust.